go/vt/sqlparser/predicate_rewriting_test.go

```diff
@@ -132,3 +132,35 @@ func TestRewritePredicate(in *testing.T) {
 		})
 	}
 }
+
+func TestExtractINFromOR(in *testing.T) {
+	tests := []struct {
+		in       string
+		expected string
+	}{{
+		in:       "(A and B) or (B and A)",
```

Comment:
add one more test case
`(a = 5  and B) or A`

Reply:
Added it. It doesn't find any additional predicates to use

Reply:
got it. thanks